source/common/stream_info/stream_info_impl.h

```diff
@@ -279,6 +279,12 @@ struct StreamInfoImpl : public StreamInfo {
 
   const Http::RequestHeaderMap* getRequestHeaders() const override { return request_headers_; }
 
+  void setResponseHeaders(const Http::ResponseHeaderMap& headers) override {
+    response_headers_ = &headers;
+  }
+
+  const Http::ResponseHeaderMap* getResponseHeaders() const override { return response_headers_; }
```

Comment:
I actually doubt that could we add this method. Long time ago, I did some discussion with other maintainers, we agreed on that the stream info should own the data it store and we finally will remove the `request_headers_` here.

Reply:
See https://github.com/envoyproxy/envoy/pull/21932#issuecomment-1202581791

And cc @adisuissa

Reply:
it seems that ,we can change CustomTagContext add response header? but can not add repsonse header in stream info ?

Reply:
I think so, according to the previous discussion or comments.

Reply:
yeah ,if previous discussion has fixed conclusion,i will try another way like struct CustomTagContext {
  const TraceContext* trace_context;
  const StreamInfo::StreamInfo& stream_info;
  const  http::ResponseHeaderMap
};   is that ok?

Reply:
Please add a detailed comment to tell this is only used for HTTP and is a special case.

Reply:
ok, i will try to add,can reopen this pr?